sunpy/map/mapbase.py

```diff
@@ -631,25 +631,15 @@ def observer_coordinate(self):
         """
         The Heliographic Stonyhurst Coordinate of the observer.
         """
+        warning_message = ""
         for keys, kwargs in self._supported_observer_coordinates:
             if all([k in self.meta for k in keys]):
                 return SkyCoord(obstime=self.date, **kwargs).heliographic_stonyhurst
+            elif any([k in self.meta for k in keys]) and not set(keys).isdisjoint(self.meta.keys()):
+                    warning_message += "For frame '{:s}' the following metadata is missing : ".format(str(kwargs['frame'])) + \
```

Comment:
I suggest you create a structure of some kind to store the frame and the missing keys here and then build the warning strings and issue the warnings at the same place lower in the function.

Reply:
Also, you should use f-strings to clean this up a little.

Reply:
Done